packages/eslint-plugin-react-hooks/src/PreferUseStateLazyInitialization.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        "Disallow function calls in useState that aren't wrapped in an initializer function",
+      recommended: false,
+      url: null,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      useLazyInitialization:
+        'To prevent re-computation, consider using lazy initial state for useState calls that involve function calls. Ex: useState(() => getValue())',
+    },
+  },
+
+  create(context) {
```

Comment:
I think code will be clearer if you just ban all calls inside useState

```suggestion
    create: (context) => {
        let useStateCallExpression = null;

        return {
            CallExpression(node) {
                if (node.callee.type === 'Identifier' && node.callee.name === 'useState') {
                    useStateCallExpression = node;
                    return;
                }

                if (useStateCallExpression) {
                    context.report({ node, messageId: 'useLazyInitialization' });
                }
            },

            'CallExpression:exit': function (node) {
                if (node === useStateCallExpression) {
                    useStateCallExpression = null;
                }
            },
        };
    },
```